### PR TITLE
Integration Tests to Docker PostgreSQL 13 / Maven Dependency Upgrades

### LINKED ISSUE #130: Migrate Integration Tests to Use PostgreSQL 13
We should upgrade our integration test setup to use the latest [PostgreSQL 13](https://www.postgresql.org/about/news/2077/) release.

Currently we are using PostgreSQL 10.6 via [db-test-utils](https://github.com/hhund/db-test-utils) -> [otj-pg-embedded](https://github.com/opentable/otj-pg-embedded). The Project otj-pg-embedded hast not been maintained in the last 12 months.

We might want to use a PostgreSQL docker container during testing. Maven setup could be similar to

https://the-codeslinger.com/2020/04/26/integration-testing-with-docker-maven-plugin-postgresql-flyway/

The GitHub Actions Setup could be similar to

https://docs.github.com/en/actions/guides/creating-postgresql-service-containers

### LINKED ISSUE #131: Upgrade Dependencies
* HAPI FHIR 5.0.2 to 5.1.0
* Camunda 7.13.0 to 7.14.0
* Jetty 9.4.30.v20200611 to 9.4.32.v20200930
* Jersey 2.31 to 2.32
* Spring 5.2.7.RELEASE to 5.2.9.RELEASE
* ...

### PR DESCRIPTION
**Pull request by hhund**
* Modifies integration Jetty and DB tests to use docker postgres13 container during maven build
* JUnit tests are executed during the maven test phase, JUnit integration tests are executed during the integration-test phase
* Upgrades Maven dependencies to latest compatible versions
* Adds bat and sh script to start postgres13 docker container to run dao and jetty integration tests manually

_As a side effect the maven build time is reduced from 26m to about 8m_

closes #130 
closes #131

### CODE REVIEW

File: dsf-fhir/dsf-fhir-server/pom.xml [hunk 1]
```diff
@@ -191,13 +191,80 @@
...
+					</execution>
+					<execution>
+						<id>stop-postgres</id>
+						<phase>post-integration-test</phase>
+						<goals>
+							<goal>stop</goal>
+						</goals>
+					</execution>
+				</executions>
 			</plugin>
```
Comment (wetret) [inline dsf-fhir/dsf-fhir-server/pom.xml:269]: Remove whitespace

File: dsf-fhir/dsf-fhir-server/src/test/java/org/highmed/dsf/fhir/profiles/OrganizationProfileTest.java [hunk 2]
```diff
@@ -89,7 +89,7 @@ public void testOrganizationProfileNotValid2() throws Exception
 		result.getMessages().stream().map(m -> m.getLocationString() + " " + m.getLocationLine() + ":"
 				+ m.getLocationCol() + " - " + m.getSeverity() + ": " + m.getMessage()).forEach(logger::info);
 
-		assertEquals(1, result.getMessages().stream().filter(m -> ResultSeverityEnum.ERROR.equals(m.getSeverity())
+		assertEquals(2, result.getMessages().stream().filter(m -> ResultSeverityEnum.ERROR.equals(m.getSeverity())
```
Comment (wetret) [inline dsf-fhir/dsf-fhir-server/src/test/java/org/highmed/dsf/fhir/profiles/OrganizationProfileTest.java:92]: Why did it change from 1 to 2?

File: dsf-fhir/dsf-fhir-server/pom.xml (hunk 1 above)
Comment (wetret) [inline dsf-fhir/dsf-fhir-server/pom.xml:269]: Maybe remove whitespace?

**Review by wetret — APPROVED** [review]
Very nice that the build time is reduced significantly.
Maybe check the 2 comments.

File: dsf-fhir/dsf-fhir-server/src/test/java/org/highmed/dsf/fhir/profiles/OrganizationProfileTest.java (hunk 2 above)
Comment (hhund) [reply dsf-fhir/dsf-fhir-server/src/test/java/org/highmed/dsf/fhir/profiles/OrganizationProfileTest.java:92]: Something changed in the HAPI validation code from 5.0.2 to 5.1.0. The testOrganizationProfileNotValid2 test case now produces two errors: Organization.type[0].coding[0].code missing and Organization.type[0].coding[0].code not being part of ttp://highmed.org/fhir/ValueSet/organization-type.

File: dsf-fhir/dsf-fhir-server/pom.xml (hunk 1 above)
Comment (hhund) [reply dsf-fhir/dsf-fhir-server/pom.xml]: done, see a47df99

**Comment by hhund** [comment]
> Very nice that the build time is reduced significantly.

Build times could further be improved. Currently every Jetty integration test class generates a number of X509 certificates and starts and stops a Jetty server. The server only needs to be started once per build and the certificates could even be reused between builds unless `mvn clean` is called. We might even want to think about using a docker container to start the Jetty server since we now depend on docker during the build anyways.

I created #134 to track this.

**Pull request merged**